ui/pages/confirmations/hooks/alerts/useTokenTrustSignalAlerts.ts

```diff
@@ -0,0 +1,88 @@
+import { useMemo } from 'react';
+import {
+  SimulationTokenStandard,
+  TransactionMeta,
+} from '@metamask/transaction-controller';
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../helpers/constants/design-system';
+import { RowAlertKey } from '../../../../components/app/confirm/info/row/constants';
+import { useConfirmContext } from '../../context/confirm';
+import { TrustSignalDisplayState } from '../../../../hooks/useTrustSignals';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+// eslint-disable-next-line import/no-restricted-paths
+import { isSecurityAlertsAPIEnabled } from '../../../../../app/scripts/lib/ppom/security-alerts-api';
+import { useTokenTrustSignalsForAddresses } from '../../../../hooks/useTokenTrustSignals';
+
+const EMPTY_ALERTS: Alert[] = [];
+const EMPTY_ACTIONS: Alert['actions'] = [];
+
+export function useTokenTrustSignalAlerts(): Alert[] {
+  const t = useI18nContext();
+  const { currentConfirmation } = useConfirmContext();
+
+  const txMeta = currentConfirmation as TransactionMeta | undefined;
+  const chainId = txMeta?.chainId;
+  const tokenBalanceChanges = txMeta?.simulationData?.tokenBalanceChanges;
+
+  const incomingTokenAddresses = useMemo<string[] | undefined>(() => {
+    if (!tokenBalanceChanges?.length) {
+      return undefined;
+    }
+    const addresses = tokenBalanceChanges
+      .filter(
+        (change) =>
+          !change.isDecrease &&
+          change.standard === SimulationTokenStandard.erc20 &&
+          change.address,
+      )
+      .map((change) => change.address.toLowerCase());
+    return addresses.length ? addresses : undefined;
+  }, [tokenBalanceChanges]);
+
+  const tokenTrustSignals = useTokenTrustSignalsForAddresses(
+    chainId,
+    incomingTokenAddresses,
+  );
+
+  return useMemo(() => {
+    if (!isSecurityAlertsAPIEnabled() || !incomingTokenAddresses?.length) {
```

Comment:
Why are we checking `isSecurityAlertsAPIEnabled` at this point? This is not firing the network requests, so at this point is more, do we have in the cache already? And is the cache still valid? If yes, then lets just render the alerts.
We should not couple the alerts being displayed to the securityAlertsApi being enabled. That perference dictates if we can call the security alerts API, not if the user will see alerts or not.